core/src/main/scala/chisel3/experimental/SerializableModuleGenerator.scala

```diff
@@ -80,4 +90,17 @@ case class SerializableModuleGenerator[M <: SerializableModule[P], P <: Serializ
 
   /** elaborate a module from this generator. */
   def module(): M = construct
+
+  /** get the definition from this generator. */
+  def definition(): Definition[M] = Builder.contextCache
+    .getOrElseUpdate(
+      SerializableModuleGenerator.CacheKey(
+        parameter,
+        implicitly[universe.TypeTag[M]]
+      ),
+      Definition.do_apply(construct.asInstanceOf[M with BaseModule])(UnlocatableSourceInfo)
```

Comment:
`[M with BaseModule]` is dirty since `BaseModule` doesn't allowed to be extended from a trait.
In runtime, this is always safe.